packages/node/src/plugins/autostaker/AutostakerPlugin.ts

```diff
@@ -0,0 +1,178 @@
+        case 'stake':
+            return _operatorContractUtils.stake
+        case 'unstake':
+            return _operatorContractUtils.unstake
+        default:
+            throw new Error('assertion failed')
+    }
+}
+
+export class AutostakerPlugin extends Plugin<AutostakerPluginConfig> {
+
+    private abortController: AbortController = new AbortController()
+
+    async start(streamrClient: StreamrClient): Promise<void> {
+        logger.info('Start autostaker plugin')
+        scheduleAtApproximateInterval(async () => {
+            try {
+                await this.runActions(streamrClient)
+            } catch (err) {
+                logger.warn('Error while running autostaker actions', { err })
+            }
+        }, this.pluginConfig.runIntervalInMs, 0.1, false, this.abortController.signal)
+    }
+
+    private async runActions(streamrClient: StreamrClient): Promise<void> {
+        logger.info('Run autostaker analysis')
+        const provider = (await streamrClient.getSigner()).provider
+        const operatorContract = _operatorContractUtils.getOperatorContract(this.pluginConfig.operatorContractAddress)
+            .connect(provider)
+        const stakeableSponsorships = await this.getStakeableSponsorships(streamrClient)
+        const stakes = await this.getStakes(streamrClient)
+        const stakedAmount = await operatorContract.totalStakedIntoSponsorshipsWei()
+        const unstakedAmount = (await operatorContract.valueWithoutEarnings()) - stakedAmount
+        logger.debug('Analysis state', {
+            stakeableSponsorships: [...stakeableSponsorships.entries()].map(([sponsorshipId, config]) => ({
+                sponsorshipId,
+                payoutPerSec: formatEther(config.payoutPerSec)
+            })),
+            stakes: [...stakes.entries()].map(([sponsorshipId, amount]) => ({
+                sponsorshipId,
+                amount: formatEther(amount)
+            })),
+            balance: {
+                unstaked: formatEther(unstakedAmount),
+                staked: formatEther(stakedAmount)
+            }
+        })
+        const actions = adjustStakes({
+            operatorState: {
+                stakes,
+                unstakedAmount
+            },
+            operatorConfig: {
+                operatorContractAddress: this.pluginConfig.operatorContractAddress,
+                minTransactionAmount: parseEther(String(this.pluginConfig.minTransactionDataTokenAmount)),
+                maxSponsorshipCount: this.pluginConfig.maxSponsorshipCount
+            },
+            stakeableSponsorships,
+            environmentConfig: {
+                minStakePerSponsorship: 5000000000000000000000n  // TODO read from The Graph (network.minimumStakeWei)
+            }
+        })
+        const signer = await streamrClient.getSigner()
+        for (const action of actions) {
+            logger.info(`Action: ${action.type} ${formatEther(action.amount)} ${action.sponsorshipId}`)
```

Comment:
Logging guidelines would suggest something like:
```suggestion
            logger.info(`Execute action: ${action.type} ${formatEther(action.amount)} ${action.sponsorshipId}`)
```